research/conditional/moe_layers/expert_choice.py

```diff
@@ -59,26 +67,48 @@ def __init__(
         ).requires_grad_(True)
         self.ln = LayerNorm(dmodel)
         assert softmax_over in ["tokens", "experts"]
+        assert not self.softmax_ungrouped or self.group_by_batch
```

Comment:
dlaczego ten if? potencjalnie można by ustawić niezależnie te rzeczy. Potencjalnie ustawiając oba na "None" domyślnie, i jeśli jeden jest ustawiony to ustawiać oba.

Reply:
tak spk tak lepiej